beacon_node/rest_api/src/beacon.rs

```diff
@@ -123,6 +128,50 @@ pub fn get_block_root<T: BeaconChainTypes>(
     ResponseBuilder::new(&req)?.body(&root)
 }
 
+fn make_sse_response_chunk(new_head_hash: SignedBeaconBlockHash) -> std::io::Result<Chunk> {
+    let mut buffer = Vec::new();
+    {
+        let mut sse_message = uhttp_sse::SseMessage::new(&mut buffer);
+        let untyped_hash: Hash256 = new_head_hash.into();
+        write!(sse_message.data()?, "{:?}", untyped_hash)?;
+    }
+    let chunk: Chunk = buffer.into();
+    Ok(chunk)
+}
+
+pub fn stream_forks<T: BeaconChainTypes>(
+    log: Logger,
+    mut events: BusReader<SignedBeaconBlockHash>,
+) -> ApiResult {
+    let (mut sender, body) = Body::channel();
+    std::thread::spawn(move || {
+        while let Ok(new_head_hash) = events.recv() {
+            let chunk = match make_sse_response_chunk(new_head_hash) {
+                Ok(chunk) => chunk,
+                Err(e) => {
+                    error!(log, "{}", e);
+                    sender.abort();
+                    break;
+                }
+            };
+            if let Err(chunk) = sender.send_data(chunk) {
+                warn!(log, "Couldn't stream chunk {:?}; entering busy loop", chunk);
+                let bytes: Bytes = chunk.into_bytes();
+                while let Err(_) = sender.send_data(bytes.clone().into()) {}
```

Comment:
This `while let Err(_)` smells a bit funky to me; if an error is never resolved we loop forever. I'm guessing you've good some good reasoning behind this.

Reply:
I hammered the the code a bit more in a local testnet and the it did enter an infinite loop.  Got rid of this in the latest push.